tests/sles4sap/publiccloud/qesap_terraform.pm

```diff
@@ -0,0 +1,180 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Deploy public cloud infrastructure using terraform using qe-sap-deployment project.
+# https://github.com/SUSE/qe-sap-deployment
+
+# Available OpenQA parameters:
+# QESAP_HANA_HA_ENABLED - Enables HA/Hana cluser scenario
+# NODE_COUNT - number of nodes to deploy. Needs to be >1 for cluster usage.
+# QESAP_VM_SIZE - VM size, sets terraform 'vm_size' parameter
+# USE_SAPCONF - (true/false) set 'false' to use saptune
+# HANA_OS_MAJOR_VERSION - sets 'hana_os_major_version' terraform parameter - default is taken from 'VERSION'
+# FENCING_MECHANISM - (sbd/native) choose fencing mechanism
+# QESAP_SCC_NO_REGISTER - define variable in openqa to skip SCC registration via ANSIBLE
+# HANA_MEDIA - Hana install media directory
+# _HANA_MASTER_PW (mandatory) - Hana master PW (secret)
+# INSTANCE_SID - SAP Sid
+# INSTANCE_ID - SAP instance id
+
+
+use base 'sles4sap_publiccloud_basetest';
+use strict;
+use warnings;
+use testapi;
+use Mojo::File 'path';
+use publiccloud::utils;
+use qesapdeployment;
+use sles4sap_publiccloud_lib;
+use serial_terminal 'select_serial_terminal';
+
+sub test_flags {
+    return {fatal => 1, publiccloud_multi_module => 1};
+}
+
+=head2 set_var_output
+
+    Check if requested openQA variable is defined and returns it's current value.
+    If the variable is not defined, it will set up the default value provided and returns it.
+
+    $variable - variable to check against OpenQA settings
+    $default - default value to set in case the variable is missing.
+
+=cut
+
+sub set_var_output {
+    my ($variable, $default) = @_;
+    return (get_var($variable)) if get_var($variable);
+    set_var($variable, $default);
+    return ($default);
+}
```

Comment:
I think we talked about it. I wanted to avoid setting variables even if they are already set with the same value. But if it is not an issue, I will use the shorter version.

Reply:
Done and pushed.